src/usr/lib/rsetup/cli/u-boot-menu.sh

```diff
@@ -136,28 +136,46 @@ rebuild_overlays() {
 enable_overlays() {
     __parameter_count_at_least_check 1 "$@"
 
-    local enable_overlays_list i
+    local enable_overlays_list i directory_name file_name have_illegal_parameters=false
 
     load_u-boot_setting
 
     for i in "$@"
     do
-        i="${i%.disabled}"
-        if [[ -e "$U_BOOT_FDT_OVERLAYS_DIR/$i.disabled" ]]
-        then
-            enable_overlays_list+=("$U_BOOT_FDT_OVERLAYS_DIR/$i")
-        elif [[ -e "$i.disabled" && "$i" == "$U_BOOT_FDT_OVERLAYS_DIR"* ]]
+        directory_name="$(dirname "$i")"
+        file_name="$(basename "$i")"
+
+        if [[ "$directory_name" == "." ]]
         then
-            enable_overlays_list+=("$i")
-        elif [[ -e "$i" && "$i" == "$U_BOOT_FDT_OVERLAYS_DIR"* ]] || [[ -e "$U_BOOT_FDT_OVERLAYS_DIR/$i" ]]
+            directory_name="$(realpath "$U_BOOT_FDT_OVERLAYS_DIR")"
+        fi
+        echo "$directory_name/$file_name"
+        if [[ "$(realpath "$directory_name")" == "$(realpath "$U_BOOT_FDT_OVERLAYS_DIR")" ]]
```

Comment:
```suggestion
        if [[ "$(realpath "$directory_name")" != "$(realpath "$U_BOOT_FDT_OVERLAYS_DIR")" ]]
```
然后打印错误提示，设置input_flag为false，然后用continue关键字来跳过后面的判断，让循环继续检查下一个文件